### PR TITLE
feat(sfu): Make session and peers interfaces

### PR DESCRIPTION
**Pull request by OrlandoCo**
#### Description

This PR will allow users to build their own peers and sessions allowing them to customize their use case.

### CODE REVIEW

File: pkg/sfu/peer.go [hunk 1]
```diff
@@ -24,28 +24,36 @@ var (
...
+	// If true it will Relay all the published tracks of the peer
 	Relay bool
 }
 
-// SessionProvider provides the session to the sfu.Peer{}
+// SessionProvider provides the session to the sfu.Peer
 // This allows the sfu.SFU{} implementation to be customized / wrapped by another package
 type SessionProvider interface {
-	GetSession(sid string) (*Session, WebRTCTransportConfig)
+	GetSession(sid string) (*session, WebRTCTransportConfig)
```
Comment (billylindeman) [inline pkg/sfu/peer.go:48]: Making session private there breaks the purpose of this interface.  I don't quite understand how this is meant to be used.  Shouldn't this be using the Session interface?

**Review by billylindeman — CHANGES_REQUESTED** [review]

File: pkg/sfu/peer.go [hunk 2]
```diff
@@ -132,13 +138,13 @@ func (p *Peer) Join(sid, uid string, config ...JoinConfig) error {
...
-		p.publisher, err = NewPublisher(p.session, uid, conf.Relay, cfg)
+		p.publisher, err = NewPublisher(p.session, uid, conf.Relay, &cfg)
 		if err != nil {
 			return fmt.Errorf("error creating transport: %v", err)
 		}
-		for _, dc := range p.session.datachannels {
+		for _, dc := range p.session.GetDCMiddlewares() {
 			if err := p.subscriber.AddDatachannel(p, dc); err != nil {
-				return fmt.Errorf("error setting subscriber default dc datachannel")
+				return fmt.Errorf("error Setting subscriber default dc datachannel")
```
Comment (tarrencev) [inline pkg/sfu/peer.go:147]:
```suggestion
				return fmt.Errorf("setting subscriber default dc datachannel: %w", err)
```

File: pkg/sfu/peer.go [hunk 3]
```diff
@@ -24,28 +24,36 @@ var (
...
 // This allows the sfu.SFU{} implementation to be customized / wrapped by another package
 type SessionProvider interface {
-	GetSession(sid string) (*Session, WebRTCTransportConfig)
+	GetSession(sid string) (Session, WebRTCTransportConfig)
 }
 
-// Peer represents a pair peer connection
-type Peer struct {
+// PeerImpl represents a pair peer connection
+type PeerImpl struct {
```
[the remark this answers is not in the document]
Comment (tarrencev) [reply pkg/sfu/peer.go]: Ah i see you changed to `PeerLocal`

**Review by tarrencev — APPROVED** [review]
lgtm. some minor naming/error handling suggestions

File: pkg/sfu/peer.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tarrencev) [reply pkg/sfu/peer.go]: Yeah i think that is fine 👍

**Pull request merged**